bootstrap/pkg/apis/apps/kfconfig/types.go

```diff
@@ -514,23 +516,43 @@ func (c *KfConfig) SyncCache() error {
 	return nil
 }
 
-// GetSecret returns the specified secret or an error if the secret isn't specified.
-func (c *KfConfig) GetSecret(name string) (string, error) {
+// Base64Encrypt encrypts a string to based64 encoding.
+func Base64Encrypt(password string) (string, error) {
+	passwordHash, err := bcrypt.GenerateFromPassword([]byte(password), 10)
+	if err != nil {
+		return "", &kfapis.KfError{
+			Code:    int(kfapis.INVALID_ARGUMENT),
+			Message: fmt.Sprintf("Error when hashing password: %v", err),
+		}
+	}
+	encodedPassword := base64.StdEncoding.EncodeToString(passwordHash)
+
+	return encodedPassword, nil
+}
+
+// GetEncodedSecret returns the encoded secret value or an error if the secret isn't specified.
+func (c *KfConfig) GetEncodedSecret(name string) (string, error) {
```

Comment:
Per above comment I don't think we want to do this. This will cause problems. Code that is calling GetSecret are expecting the raw secret not a bcrypted value.

Reply:
I will move the logic to build basic auth secret.